src/open_samus_returns_rando/lua_editor.py

```diff
@@ -277,11 +277,14 @@ def chunks(array: list[str], n: int) -> Iterable[list[str]]:
             )
         )
 
-        if "required_dna" in objective:
-            required_dna = objective["required_dna"]
-        else:
-            starting_dna = [item for item in inventory if item.startswith("ITEM_RANDO_DNA")]
-            required_dna = 39 - len(starting_dna)
+        starting_dna = [item for item in inventory if item.startswith("ITEM_RANDO_DNA")]
+        total_dna = objective.get("total_dna", 39)
+        placed_dna = objective.get("placed_dna", 0)
+        objective_dna = objective.get("required_dna", 0)
+
+        # Handle starting with extra DNA which would affect the objective
+        additional_dna = placed_dna - (total_dna - len(starting_dna))
+        required_dna = objective_dna - additional_dna
```

Comment:
I don't want to be a annoying and probably nobody cares about the backward compatibility of patcher.json files but
```
placed_dna = objective.get("placed_dna", 0)
```
is a bad default choice.  That would lead to `additional_dna = 0 - (39 - len(starting_dna))`, so it will be negative and the next line will raise the `required_dna` to a value higher than the collectible DNA.
We cannot fix old patcher.json files because we require the new infomraiton to fix it but to not break the old working patcher.json files without random starting DNA, we could force `additional_dna` to be 0.
E.g. by if-else logic or by make it mathematical always be zero as default `placed_dna = objective.get("placed_dna", -1 * (total_dna - len(starting_dna))`

Reply:
That's fine. Ill make the change

Reply:
Brain fart by me, but luckily you copied the missing closing `)` and the tests fail. It must be
```
placed_dna = objective.get("placed_dna", total_dna - len(starting_dna))
```

Reply:
Fixed